tests/base.py

```diff
@@ -0,0 +1,417 @@
+                    'assigned_to',
+                    'is_dateless',
+                    'object_id',
+                    'opportunity_value_formatted',
+                    'updated_by_name',
+                    'phone_formatted',
+                    'created_by_name',
+                    'lead_name',
+                    'assigned_to_name',
+                    'opportunity_value_period',
+                },
+            'users': {
+                    'date_updated',
+                    'id',
+                    'last_name',
+                    'date_created',
+                    'organizations',
+                    'email',
+                    'first_name',
+                    'image',
+                    'last_used_timezone',
+                },
+        }
+        return auto_fields
+
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(logging="Ensuring environment variables are sourced.")
+        missing_envs = [
+            x for x in [
+                'TAP_CLOSEIO_API_KEY' #TODO add account_id and password if needed(credentials)
+            ] if os.getenv(x) is None
+        ]
+
+        if len(missing_envs) != 0:
+            raise Exception("Missing environment variables: {}".format(missing_envs))
+
+
+    ##########################################################################
+    ### Tap Specific Methods
+    ##########################################################################
+
+
+    @staticmethod
+    def expected_pagination_fields(): # TODO does this apply?
+        return {
+            "Test Report 1" : set(),
+            "Audience Overview": {
+                "ga:users", "ga:newUsers", "ga:sessions", "ga:sessionsPerUser", "ga:pageviews",
+                "ga:pageviewsPerSession", "ga:sessionDuration", "ga:bounceRate", "ga:date",
+                # "ga:pageviews",
+            },
+            "Audience Geo Location": set(),
+            "Audience Technology": set(),
+            "Acquisition Overview": set(),
+            "Behavior Overview": set(),
+            "Ecommerce Overview": set(),
+        }
+
+
+    # TODO refactor code to remove this method if possible, ga4 relic
+    def get_stream_name(self, stream):
+        """
+        Returns the stream_name given the tap_stream_id because synced_records
+        from the target output batches records by stream_name
+
+        Since the GA4 tap_stream_id is a UUID instead of the usual case of
+        tap_stream_id == stream_name, we need to get the stream_name that
+        maps to tap_stream_id
+        """
+        stream_name=stream
+        # custom_reports_names_to_ids().get(tap_stream_id, tap_stream_id)
+        return stream_name
+
+    def perform_and_verify_table_and_field_selection(self,
+                                                     conn_id,
+                                                     test_catalogs,
+                                                     select_all_fields=True):
```

Comment:
This doesn't feel very Close IO specific. Should this be moved into the base suite?